docs/handling-eviction-requests.md

```diff
@@ -0,0 +1,79 @@
+| LiveMigrateIfPossible | True              | True                       | True                          | 200 - Eviction granted |
+| LiveMigrateIfPossible | False             | False                      | True                          | 200 - Eviction granted |
+| External              | N/A               | True                       | True                          | 200 - Eviction granted |
+
+The webhook sets the value of the VMI's `Status.EvacuationNodeName` to the node name its currently running on, so the evacuation controller will know it needs to migrate it to another node.
+
+# Pod Distribution Budget
+In order for the eviction of VMIs to happen in a controlled manner, KubeVirt protects part of the `virt-launcher` pods with a PDB which blocks eviction requests.
+
+`virt-controller` has a `Disruption Budget Controller` which decides whether a `virt-launcher` pod should be protected based on the eviction strategy of its controlling VMI:
+
+| Eviction Strategy     | Is a PDB required             | 
+|-----------------------|-------------------------------|
+| None                  | False                         |
+| LiveMigrate           | True                          |
+| LiveMigrateIfPossible | Only if the VMI is migratable |
+| External              | True                          |
+
+# Eviction Approval Summery
+
+The eviction request's initiator will observe one of the following responses:
+
+| Eviction Strategy     | Is VMI migratable | Is VMI marked for eviction | Does Webhook approve eviction | Does PDB allow eviction | Response                         |
+|-----------------------|-------------------|----------------------------|-------------------------------|-------------------------|----------------------------------|
+| None                  | N/A               | False                      | True                          | True                    | 200 - Eviction granted           |
+| LiveMigrate           | True              | True                       | True                          | False                   | 429 - Eviction blocked by PDB    |
+| LiveMigrate           | False             | False                      | False                         | False                   | 429 - Eviction denied by webhook |
+| LiveMigrateIfPossible | True              | True                       | True                          | False                   | 429 - Eviction blocked by PDB    |
+| LiveMigrateIfPossible | False             | False                      | True                          | True                    | 200 - Eviction granted           |
+| External              | N/A               | True                       | True                          | False                   | 429 - Eviction blocked by PDB    |
+
+To summarize:
+1. The eviction request is granted only if both the webhook and the PDB allow them.
```

Comment:
Why do we need two mechanisms? Why pdb is not enough?

Reply:
The PDB just blocks the eviction request. I think it is still in-use as a precaution in case the webhook is unavailable, since the webhook is configured to be ignored in case of failure (failurePolicy: Ignore)[1].

The webhook marks the VMI for eviction.

[1] https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy

Reply:
The webhook cannot give the same as protection the PDB - which also protects two pods during migrations.

Reply:
Would you explain why two different mechanisms are needed? What is the problems and benefits of each of them? Please include the answer in the text of this documentation.

Reply:
Please see the note under `Pod Eviction Webhook`, it should answer this question.

Reply:
Thanks.